src/events/balancer.py

```diff
@@ -195,16 +196,19 @@ def rebalance(self, event: ActionEvent) -> None:
                 not self.balancer_manager.cruise_control.ready,
                 "CruiseControl balancer service has not yet collected enough data to provide a partition reallocation proposal",
             ),
+            (
+                event.params.get("brokerid", None) is None
+                and event.params["mode"] != RebalanceMode.FULL,
+                "'add' and 'remove' rebalance action require passing the 'brokerid' parameter",
```

Comment:
**suggestion:** I might consider explicitly saying:

```python
if event.params["mode"] in [RebalanceMode.ADD, RebalanceMode.REMOVE]
and event.params.get("brokerid", None) is None
```

The explicit listing of the 'modes that need it', and the failure conditional coming after reads a bit better imo.

Reply:
**thought:** `0` being Falsey is annoying as heck

Reply:
Made this condition more explicit in b48ecec153d6. Closing this thread, but I'll revisit if we change what `RebalanceMode` is